Source/JavaScriptCore/b3/B3ReduceStrength.cpp

```diff
@@ -1884,13 +1884,11 @@ class ReduceStrength {
                 }
 
                 // Turn this: SExt8To64(BitAnd(input, mask)) where (mask & 0x80) == 0
-                // Into this: ZExt32(BitAnd(input, const & 0x7f))
+                // Into this: ZExt32(BitAnd(input, mask & 0x7f))
                 if (!(mask & 0x80)) {
-                    replaceWithNew<Value>(ZExt32, m_value->origin(),
-                        m_proc.add<Value>(
-                            BitAnd, m_value->origin(), input,
-                            m_insertionSet.insert<Const32Value>(
-                                m_index, m_value->origin(), mask & 0x7f)));
+                    Const32Value* maskValue = m_insertionSet.insert<Const32Value>(m_index, m_value->origin(), mask & 0x7f);
+                    Value* bitAndValue = m_insertionSet.insert<Value>(m_index, BitAnd, m_value->origin(), input, maskValue);
+                    replaceWithNew<Value>(ZExt32, m_value->origin(), bitAndValue);
```

Comment:
Why do we need the ZExt32? Isn’t that guaranteed from the BitAnd constant?

Reply:
Oh duh because the left operand is I32. Never mind me.

Reply:
This is because B3::Value is typed. Since `m_value->child(0)->child(1)->hasInt32()`, that `BitAnd` is `Int32`. But `SExt8To64` wants a `Int64` result.